enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. Cluster Creator creates a ClusterDeployment describing a new cluster. It describes how many control-plane and worker agents to expect. It also includes a label selector to match Agent resources that should be part of the cluster.
+1. Cluster Creator applies a label to Agents if necessary so that they match the ClusterDeployment's selector.
+1. Once there are enough ready Agents of each role to fulfill the expected number as expressed on the ClusterDeployment, installation begins.
+
+#### Static Networking
+
+Some customers have asked for the ability to provide static network details
+up-front for each host instead of using DHCP. They want to define this
+configuration at the same time they define the corresponding BareMetalHost.
+
+A new resource called NMStateConfig will have a Spec with the following
+fields:
+
+* MACAddress: a MAC address for any network device on the host to which this config should be applied. This value is only used to ensure that the config is applied to the intended host.
+* Config: a byte array that can contain a raw [nmstate](https://www.nmstate.io/) network config.
+
+nmstate is already in use within OpenShift for applying network configuration
+to nodes via the [kubernetes-nmstate
+operator](https://github.com/nmstate/kubernetes-nmstate). It uses a similar
+pattern of embedding a raw nmstate yaml structure as a byte array in the Spec.
+
+Each NMStateConfig resource will have a label that corresponds to a InstallEnv.
+The raw YAML configs for each matching resource will be rendered to a network
+config by the Assisted Service and then embedded into the discovery ISO for
+that InstallEnv. At runtime, the discovery ISO will find the config that
+matches a MAC address on the current host and then apply the config. It does
+not matter which interface has the matching MAC address; the matching is merely
+used to identify that the current host corresponds to a given config.
+
+The NMStateConfig resource design is being discussed [in a
+pull request](https://github.com/openshift/assisted-service/pull/969).
+
+#### Install Device
+
+The Agent resource Spec will include a field on which to specify the storage device
+where the OS should be installed. That field must be set by a platform-specific
+controller or some other actor that understands the underlying host.
+
+For bare metal, the BareMetalHost resource Spec already includes a
+RootDeviceHints section that will be utilized. The new Baremetal Agent
+Controller (previously described in the "Create Cluster" scenario) will use the
+BareMetalHost's RootDeviceHints and the Agent's discovery data to populate this
+field on the Agent.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke cluster.
```

Comment:
I feel like I'm missing something obvious here - where exactly does the user specify which cluster this worker should be added to?

Reply:
You're right, this text assumes a ClusterDeployment exists.  This can happen either by the cluster having been installed by Hive or by importing/adopting it into Hive.

Reply:
Oops. I'm adding clarification that:
* this assumes the ClusterDeployment exists
* the Agent needs a label to match a selector on the ClusterDeployment.